### PR TITLE
1066 generate pdf

### PR DESCRIPTION
**Pull request by matthopson**
This incorporates the DynamSoft library, including:
- Injecting the scripts required to perform scan operations dynamically
- Adding necessary state for management the library's presence and some basic UI
- Scan button and Select Source modal
- Function for converting image data into a PDF
- Wrapper API for the TWAIN library via applicationContext
- Supporting sequences and actions

### CODE REVIEW

File: shared/src/business/useCases/getScannerInterfaceInteractor.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    const startScanSession = () => {
+      DWObject.IfDisableSourceAfterAcquire = true;
+      DWObject.OpenSource();
+      DWObject.AcquireImage();
+    };
+
+    return {
+      DWObject,
+      changeSource: null,
```
Comment (codyseibert) [inline shared/src/business/useCases/getScannerInterfaceInteractor.js]: what is this one used for?

Comment (matthopson) [reply shared/src/business/useCases/getScannerInterfaceInteractor.js]: Good catch. I already have it removed in the stuff I'm about to push.

File: web-client/src/presenter/actions/startScanAction.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @param {Object} providers the providers object
+ * @param {Object} providers.applicationContext the application context used for getting the scanner API
+ * @param {Function} providers.props the cerebral props object used for getting the props.scannerSourceName
+ * @param {Function} providers.store the cerebral store used for setting state.path
+ */
+export const startScanAction = async ({ applicationContext, props, store }) => {
+  store.set(state.isScanning, true);
+  const scanner = applicationContext.getScanner();
+
+  scanner.setSourceByName(props.scannerSourceName);
```
Comment (codyseibert) [inline web-client/src/presenter/actions/startScanAction.js]: any clue if this throws an exception if provided a bad source name

**Review by codyseibert — APPROVED** [review]

File: web-client/src/presenter/actions/startScanAction.js (hunk 2 above)
Comment (matthopson) [reply web-client/src/presenter/actions/startScanAction.js]: That's a holdover from a previous flow. Removing this in favor of setting the source via sequence.

**Pull request merged**